### PR TITLE
Wrath Classic Phase 4 Update & Fixes

### PR DESCRIPTION
**Pull request by Anyalah**
Includes Timothy-Page Mount API Changes

- Add Defiler's Scourgestone Vendor
- Update Dalaran Vendor Rings/Gems
- #397, #388 Update to use new Mount API 
- #396 Added "Maiden of Grief" Missing items in Titan Rune Beta - Halls of Stone
- #395 Fixed "King Dred" Wrong loot in Titan Rune Beta - Drak'Tharon Keep
- #394 Fixed "The Prophet Tharon'ja" Wrong loot in Titan Rune Beta - Drak'Tharon Keep
- #393 Added Missing Leatherworking Fur Linings
- #391 Added Enchanting Cloak Resistances (Arcane, Fire, Frost, Nature, Shadow), Removed Duplicates
- #387 Fixed Incorrect Set Links for Death Knight Season 7 PvP Set  
- #381 Added "Six Shared Loot Bosses" Missing items in Titan Rune Beta - Violet Hold

### CODE REVIEW

**Comment by jazminite** [comment]
works for me, thanks! <3

**Comment by leehmanQQ** [comment]
yep works for me as well

**Comment by Remsslk** [comment]
Tysm <3

**Comment by Anyalah** [comment]
I have created a [release](https://github.com/Anyalah/AtlasLootClassic/releases/tag/v3.4.3.pre-release) here which has a direct download link until Hoizame can update
[Direct Download Link](https://github.com/Anyalah/AtlasLootClassic/releases/download/v3.4.3.pre-release/AtlasLootClassic_Phase4.zip)

**Comment by mykkenny** [comment]
Is there any way to prevent the "AtlasLoot-Dev: Player 'PlayerName-Server' sends version '3010003'!" message being spammed when using this release?

**Comment by bferg317** [comment]
Running this release just fine

File: .vs/slnx.sqlite
Comment (jakobbsm) [inline .vs/slnx.sqlite:1]: It seems this configuration file snuck itself in. Might wanna check out this .gitignore for VS Code.
https://github.com/github/gitignore/blob/main/Global/VisualStudioCode.gitignore

Comment (bferg317) [reply .vs/slnx.sqlite:1]: Definitely a good suggestion. By default my .gitignores always have these:
```
# OS Junk. Currently only Windows stuff: thumbnailer db and temp files.
**thumbs.db
~*

# IDE Junk. Currently only VSCode stuff.
.VSCode
.vscode
*.code-workspace
*.code-project
```

**Comment by BIIMSDev** [comment]
> Is there any way to prevent the "AtlasLoot-Dev: Player 'PlayerName-Server' sends version '3010003'!" message being spammed when using this release?

@mykkenny

hi, 
It was temporarily fixed.
Alarms are no longer displayed.
Test complete!

download link :) 
[https://drive.google.com/file/d/1xBvM7kd20Lb2I0wlhOqdZ3saQtyR3Uen/view?usp=sharing](url)

**Comment by ssateneth** [comment]
Can you please add #400 and #411

File: AtlasLootClassic/AtlasLootClassic_Wrath.toc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿## Interface: 30402
+﻿## Interface: 30404
```
Comment (Barov2) [inline AtlasLootClassic/AtlasLootClassic_Wrath.toc:1]:
```suggestion
﻿## Interface: 30403
```

File: AtlasLootClassic_DungeonsAndRaids/data-wrath.lua [hunk 2]
```diff
@@ -1225,42 +1225,40 @@ data["DrakTharonKeep"] = {
...
+                { 1, 45624 }, -- Emblem of Conquest 
+				{ 3, 45303 }, -- Band of Draconic Guile 
+				{ 4, 45972 }, -- Pulse Baton 
+				{ 5, 45974 }, -- Shoulderguards of Assimilation 
+				{ 7, 43995 }, -- Enamored Cowl 
+				{ 8, 39416 }, -- Kel'Thuzad's Reach 
+				{ 9, 43996 }, -- Sabatons of Firmament 
+				{ 16, "INV_Box_01", nil, AL["Tier Token iLvl200"], nil, "Tier7TitanRuneTokens" }, 
+				{ 17, "INV_Box_01", nil, AL["Tier Token iLvl225"], nil, "Tier8TitanRuneTokens" }, 
+				{ 19, 43102 }, -- Frozen Orb { 21, 41796 }, -- Design: Infused Twilight Opal
```
Comment (Barov2) [inline AtlasLootClassic_DungeonsAndRaids/data-wrath.lua:1237]:
```suggestion
 				{ 19, 43102 }, -- Frozen Orb
 				{ 21, 41795 }, -- Design: Timeless Forest Emerald
```

File: AtlasLootClassic_DungeonsAndRaids/data-wrath.lua [hunk 3]
```diff
@@ -1225,42 +1225,40 @@ data["DrakTharonKeep"] = {
...
+				{ 3, 45303 }, -- Band of Draconic Guile 
+				{ 4, 45972 }, -- Pulse Baton 
+				{ 5, 45974 }, -- Shoulderguards of Assimilation 
+				{ 7, 43995 }, -- Enamored Cowl 
+				{ 8, 39416 }, -- Kel'Thuzad's Reach 
+				{ 9, 43996 }, -- Sabatons of Firmament 
+				{ 16, "INV_Box_01", nil, AL["Tier Token iLvl200"], nil, "Tier7TitanRuneTokens" }, 
+				{ 17, "INV_Box_01", nil, AL["Tier Token iLvl225"], nil, "Tier8TitanRuneTokens" }, 
+				{ 19, 43102 }, -- Frozen Orb { 21, 41796 }, -- Design: Infused Twilight Opal 
+				{ 23, "ac1860" }, { 101, 40616 }, -- Helm of the Lost Conqueror
```
Comment (Barov2) [inline AtlasLootClassic_DungeonsAndRaids/data-wrath.lua:1238]:
```suggestion
				{ 23, "ac1860" },
				{ 101, 40616 }, -- Helm of the Lost Conqueror 
```

**Review by Barov2 — CHANGES_REQUESTED** [review]

**Comment by saimpot** [comment]
Hey @Anyalah and @Timothy-Page, I tried this PR and I have a feeling that because of a massive update that it's trying to do somewhere either it's sending amount of data to a huge number of clients, or receiving, it's cause crazy lag spikes every 40-60 or so seconds. 

I haven't been able to pinpoint this, but I had a ping running to google.com with constant 34ms while doing dungeons, and was getting these crazy lag spikes, and no other connection problems. As soon as I deleted this, it was gone.

I know it's kinda anecdotal evidence,  but maybe someone else had the same issue or you have an idea on what's happening.

**Comment by BjarkeM** [comment]
> Hey @Anyalah and @Timothy-Page, I tried this PR and I have a feeling that because of a massive update that it's trying to do somewhere either it's sending amount of data to a huge number of clients, or receiving, it's cause crazy lag spikes every 40-60 or so seconds.
> 
> I haven't been able to pinpoint this, but I had a ping running to google.com with constant 34ms while doing dungeons, and was getting these crazy lag spikes, and no other connection problems. As soon as I deleted this, it was gone.
> 
> I know it's kinda anecdotal evidence, but maybe someone else had the same issue or you have an idea on what's happening.

It's completely unrelated, none of the changes here would do that.

**Comment by saimpot** [comment]
> > Hey @Anyalah and @Timothy-Page, I tried this PR and I have a feeling that because of a massive update that it's trying to do somewhere either it's sending amount of data to a huge number of clients, or receiving, it's cause crazy lag spikes every 40-60 or so seconds.
> > I haven't been able to pinpoint this, but I had a ping running to google.com with constant 34ms while doing dungeons, and was getting these crazy lag spikes, and no other connection problems. As soon as I deleted this, it was gone.
> > I know it's kinda anecdotal evidence, but maybe someone else had the same issue or you have an idea on what's happening.
> 
> It's completely unrelated, none of the changes here would do that.

Are you sure? Not even the broadcast thing that prints to the chat "AtlasLoot-Dev: Player 'PlayerName-Server' sends version '3010003'!" message delete Ver"?

**Comment by Timothy-Page** [comment]
@saimpot I have not reviewed the code that Anyalah updated but the code I changed is also used in my add "Wrath Random Mounter" and is called everytime the zone name next to the mini map changes and it is not causing issues.

**Comment by saimpot** [comment]
> @saimpot I have not reviewed the code that Anyalah updated but the code I changed is also used in my add "Wrath Random Mounter" and is called everytime the zone name next to the mini map changes and it is not causing issues.

That's good to hear, I'll start using this again then and see if I notice anything weird. Thank you for the update and for the work that you do! Both of you!

**Comment by repsejnworb** [comment]
@Hoizame any ETA on merging?

**Comment by andreasgottwald** [comment]
> > > Hey @Anyalah and @Timothy-Page, I tried this PR and I have a feeling that because of a massive update that it's trying to do somewhere either it's sending amount of data to a huge number of clients, or receiving, it's cause crazy lag spikes every 40-60 or so seconds.
> > > I haven't been able to pinpoint this, but I had a ping running to google.com with constant 34ms while doing dungeons, and was getting these crazy lag spikes, and no other connection problems. As soon as I deleted this, it was gone.
> > > I know it's kinda anecdotal evidence, but maybe someone else had the same issue or you have an idea on what's happening.
> > 
> > 
> > It's completely unrelated, none of the changes here would do that.
> 
> Are you sure? Not even the broadcast thing that prints to the chat "AtlasLoot-Dev: Player 'PlayerName-Server' sends version '3010003'!" message delete Ver"?

It's normal if you have a dev-version - you can disable it by commenting out the atlasloot.lua line 60:

function AtlasLoot:DevPrint(msg)
	if AtlasLoot.IsDevVersion then
		-- print("|cff33ff99AtlasLoot-Dev|r: "..(msg or ""))
	end
end

**Comment by MyTechnoHunter** [comment]
> > > > Hey @Anyalah and @Timothy-Page, I tried this PR and I have a feeling that because of a massive update that it's trying to do somewhere either it's sending amount of data to a huge number of clients, or receiving, it's cause crazy lag spikes every 40-60 or so seconds.
> > > > I haven't been able to pinpoint this, but I had a ping running to google.com with constant 34ms while doing dungeons, and was getting these crazy lag spikes, and no other connection problems. As soon as I deleted this, it was gone.
> > > > I know it's kinda anecdotal evidence, but maybe someone else had the same issue or you have an idea on what's happening.
> > > 
> > > 
> > > It's completely unrelated, none of the changes here would do that.
> > 
> > 
> > Are you sure? Not even the broadcast thing that prints to the chat "AtlasLoot-Dev: Player 'PlayerName-Server' sends version '3010003'!" message delete Ver"?
> 
> It's normal if you have a dev-version - you can disable it by commenting out the atlasloot.lua line 60:
> 
> function AtlasLoot:DevPrint(msg) if AtlasLoot.IsDevVersion then -- print("|cff33ff99AtlasLoot-Dev|r: "..(msg or "")) end end

open the toc files, change from ## Version: @project-version@ to a real version number.. the code checks for the contents of that field and if found changes to a string with -dev.. which enables the dev print

if needed just use the older version number ## Version: v3.1.3

**Pull request merged**